examples/report.yaml

```diff
@@ -1,31 +1,32 @@
-defaults:
+debug:
   experiment:
     model_file: examples/output/<EXP>.mod
     hyp_file: examples/output/<EXP>.hyp
     out_file: examples/output/<EXP>.out
     err_file: examples/output/<EXP>.err
     run_for_epochs: 1
     eval_metrics: bleu
-  train:
-    trainer: adam
-    learning_rate: 0.01
-    default_layer_dim: 256 
-    dropout: 0.0
+  train: !XnmtTrainer
+    trainer: !AdamTrainer
```

Comment:
The terminology `Trainer` is now overloaded. This is a bit annoying because, `AdamTrainer` should probably be `AdamOptimizer`, but that clashes with DyNet's terminology (which I don't plan on changing just because it'd be a mess trying to maintain backward compatibility). Maybe we can rename `XnmtTrainer` something like `TrainingRegimen`?

Reply:
Sure, I can rename it to TrainingRegimen. xnmt.optimizer.AdamTrainer is currently a wrapper class around dynet.AdamTrainer (mostly because it seems not possible to serialize cython objects), so we could easily rename it to xnmt.optimizer.AdamOptimizer. What do you think? We also have training_strategy which is a bit confusing and could maybe be renamed..